packages/mui-utils/src/deepmerge/deepmerge.test.ts

```diff
@@ -132,6 +132,17 @@ describe('deepmerge', () => {
     expect(result.element).to.equal(element2);
   });
 
+  it('should not deep clone React component', () => {
+    // most 3rd-party components use `forwardRef`
+    const Link = React.forwardRef((props, ref) => React.createElement('a', { ref, ...props }));
+    const result = deepmerge(
+      { defaultProps: { component: 'a' } },
+      { defaultProps: { component: Link } },
+    );
+
+    expect(result.defaultProps.component).to.equal(Link);
```

Comment:
Without the change in this PR, this test throws because the component is a new object. This test ensure that the Link is not iterate by `deepmerge`.